Add simulation docs (#1589)
hpi-sam/digital-fuesim-manv: docs/src/3_simulation/1_general.md

# Allgemeines

## Konzepte

Über die dynamische Simulation von Patienten und technischen Herausforderungen hinaus bietet FüSim Digital die Möglichkeit, das gesamte Geschehen in einem oder mehreren Bereichen auf der Karte zu simulieren. Für jeden _simulierten Bereich_ kann dabei festgelegt werden, wie sich die dortige, simulierte Führungskraft verhalten soll. Dadurch können zum Beispiel ganze Patientenablagen oder Bereitstellungsräume simuliert werden.

> [!WARNING]
> Die simulierten Bereiche unterstützen aktuell nur Behandlung und Transport von Patienten und die Organisation von Bereitstellungsräumen und sind auf MANV-Lagen zugeschnitten. Technische Herausforderungen können von simulierten Bereichen nicht bearbeitet werden.

Diese Funktion ermöglicht es, auch in kleinen Gruppen Einsatzleitungen auszubilden, die in einer sich dynamisch entwickelnden Lage agieren müssen, ohne dass weitere Teilnehmende oder Übungsleitende für die Rollen der untergeordneten Führungskräfte benötigt werden.

> Der Parallelbetrieb von simulierten Bereichen und von Teilnehmenden besetzten Ansichten innerhalb einer Übung ist möglich. Allerdings können einzelne Elemente (Fahrzeuge, Personal, Patienten, …) zu einem bestimmten Zeitpunkt nur entweder von der Simulation oder von Teilnehmenden genutzt werden. Zur Verdeutlichung sind die simulierten Bereiche auf der Karte als ausgefüllte Flächen dargestellt, deren Inhalte nicht sichtbar sind. Eine explizite Übergabe von Fahrzeugen, Personal und Patienten zwischen Simulation und Teilnehmenden ist möglich.

Reviewer: ```suggestion
Der Parallelbetrieb von simulierten Bereichen und von durch Teilnehmende besetzten Ansichten innerhalb einer Übung ist möglich. Allerdings können einzelne Elemente (Fahrzeuge, Personal, Patienten, …) zu einem bestimmten Zeitpunkt nur entweder von der Simulation oder von Teilnehmenden genutzt werden. Zur Verdeutlichung sind die simulierten Bereiche auf der Karte als ausgefüllte Flächen dargestellt, deren Inhalte nicht sichtbar sind. Eine explizite Übergabe von Fahrzeugen, Personal und Patienten zwischen Simulation und Teilnehmenden ist möglich.
```

> Für eine möglichst natürliche Interaktion der Teilnehmenden mit einem simulierten Bereich können Informationen per Funkspruch ausgetauscht werden. Abhängig von seinen Verhaltensweisen erstellt ein simulierter Bereich selbst Funksprüche, außerdem können per Funk Befehle erteilt und Informationen abgefragt werden. FüSim Digital hat keine Anbindung an Funktechnik, stattdessen werden Funksprüche in Textform bereitgestellt. Für die Übungsdurchführung wird ein Teilnehmer oder Übungsleiter benötigt, der die Rolle des "Schnittstellenfunkers" übernimmt und die Funksprüche durchsagt oder Befehle entgegennimmt.

Reviewer: ```suggestion
Für eine möglichst natürliche Interaktion der Teilnehmenden mit einem simulierten Bereich können Informationen per Funkspruch ausgetauscht werden. Abhängig von seinen Verhaltensweisen erstellt ein simulierter Bereich selbst Funksprüche, außerdem können per Funk Befehle erteilt und Informationen abgefragt werden. FüSim Digital hat keine Anbindung an Funktechnik, stattdessen werden Funksprüche in Textform bereitgestellt. Für die Übungsdurchführung wird eine Person benötigt, die die Rolle des "Schnittstellenfunkers" übernimmt und die Funksprüche durchsagt oder Befehle entgegennimmt.
```

## Begriffe

### Simulierter Bereich

Die Simulation ist in simulierten Bereichen organisiert, die auf der Karte angeordnet werden. Jeder Bereich kann einzeln konfiguriert werden und dadurch individuelles Verhalten aufweisen.

> Simulierte Bereiche können wie Ansichten auf der Karte platziert und Fahrzeuge, Personal und Patienten hinzugefügt werden. Ihre innere räumliche Anordnung bleibt jedoch verborgen. Informationen aus der Simulation können per Funkspruch abgefragt werden.

Reviewer: ```suggestion
Simulierte Bereiche können wie Ansichten auf der Karte platziert und miteinander oder mit Ansichten verbunden werden. Per Drag-and-Drop werden Fahrzeuge, Personal und Patienten hinzugefügt. Ihre innere räumliche Anordnung bleibt jedoch verborgen. Informationen aus der Simulation können per Funkspruch abgefragt werden.
```

Author: What do you mean by connecting simulated regions to viewports?

There is no explicit mechanism that allows connecting, besides that they can be used together and thus can interact, for example through transfer points. But from your wording, I would expect that whatever kind of connection (e.g. a transfer connection) can be set up, which is not the case.

Reviewer: yeah, you are right, I mean something like:
"`können miteinander oder mit normalen Transferpunkten verbunden werden`"

Umgekehrt nimmt der Schnittstellenfunker alle Funksprüche, Zurufe und schriftliche Befehle entgegen, die an Personal aus einem simulierten Bereich gerichtet sind, und passt die Einstellungen der Simulation entsprechend an. Gängige Anfragen (zum Beispiel die Anzahl von Patienten oder Fahrzeugen) und Befehle (zum Beispiel, Fahrzeuge aus dem Bereitstellungsraum zu entsenden) kann der Schnittstellenfunker mit wenigen Klicks über eine spezielle Benutzeroberfläche bearbeiten. Für einen möglichst großen Handlungsspielraum der Einsatzleitung kann der Schnittstellenfunker auf die bekannte Übungsleiter-Benutzeroberfläche mit allen Einstellungsmöglichkeiten zurückgreifen.

Der Schnittstellenfunker muss sich selbst keine Informationen zum Übungsablauf merken oder mit der Karte interagieren, sodass eine Person als Schnittstellenfunker mehrere simulierte Bereiche bedienen kann. Für besonders große Übungen, oder wenn neben der Einsatzleitung noch weitere Teilnehmende mit simulierten Bereichen interagieren können sollen, kann es notwendig sein, die Rolle des Schnittstellenfunkers mit mehreren Personen zu besetzen, um Engpässe zu vermeiden.

## Aufbau des Simulierte-Bereich-Fensters

> Bei Klick auf einen simulierten Bereich öffnet sich ein Popup, über welches dieser Bereich konfiguriert werden kann, analog zu anderen Übungselemente auf der Karte. Da die Einstellungsmöglichkeiten von simulierten Bereichen sehr komplex sind, empfiehlt es sich aber, in die großen Simulationseinstellungen zu wechseln. Diese ist über <kbd>**Durchführung > Simulationseinstellungen (Übungsleitung)**</kbd> erreichbar.

Reviewer: ```suggestion
Bei Klick auf einen simulierten Bereich öffnet sich ein Popup, über welches dieser Bereich konfiguriert werden kann, analog zu anderen Übungselementen auf der Karte. Da die Einstellungsmöglichkeiten von simulierten Bereichen sehr komplex sind, empfiehlt es sich aber, in die großen Simulationseinstellungen zu wechseln. Diese ist über <kbd>**Durchführung > Simulationseinstellungen (Übungsleitung)**</kbd> erreichbar.
```
